landoapi/api/try_push.py

```diff
@@ -60,14 +59,6 @@ def build_revision_from_patch_helper(helper: PatchHelper, repo: Repo) -> Revisio
 
     raw_diff = helper.get_diff()
 
-    # Check diff for errors.
-    parsed_diff = rs_parsepatch.get_diffs(raw_diff)
-    errors = DiffAssessor(
-        author=author, commit_message=commit_message, parsed_diff=parsed_diff, repo=repo
-    ).run_diff_checks()
-    if errors:
-        raise ValueError(f"Patch failed checks: {' '.join(errors)}")
-
```

Comment:
It looks like this was moved to be inside `PushAssessor`. I think it would be cleaner if the two were kept separate, and their results combined after each one is called independently. This also changes the logic of when/how diff errors are raised. This might be OK but it's not explicitly called out or explained in this PR.

Reply:
Since the push assessor is iterating over each diff and performing checks, and the `DiffAssessor` is essentally doing the same thing, I think it makes sense to run the `DiffAssessor` in the same loop for now. They both end up returning error strings anyways.